### PR TITLE
ENG-2620 DAG Line Crossing Fix

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR removes elk and brings back the old positioning algorithm.

Edge crossings should be much less likely, however they might seem a little long when there is a collapsed DAG node.

## Related issue number (if any)
ENG-2620

## Loom demo (if any)
Screehshots:
image

image

image

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/utils/reactflow.ts [hunk 1]
```diff
@@ -58,14 +59,16 @@ export function getOperatorNode(
       label: op.name,
     },
     // Give an initial position. We will reposition this node later.
-    position: { x: 0, y: 0 },
+    //position: { x: 0, y: 0 },
```
Comment (agiron123) [inline src/ui/common/src/utils/reactflow.ts:62]: remove comment

**Comment by vsreekanti** [comment]
@agiron123 @likawind, it shouldn't be too difficult to modify the inputs to the algorithm to take into account metrics and checks, right? The algorithm doesn't really care about whether it's positioning an artifact or an operator, correct? If so, with a little bit of preprocessing, we should be able to "collapse" the nodes before sending the API call.

**Comment by agiron123** [comment]
> @vsreekanti I think what we can do here is add Artifacts to the request body that's sent out when doing positioning so that we can check the Artifact type and remove edge from the DAG on the backend before the positioning algorithm is run.

**Comment by agiron123** [comment]
Not sure why this linter issue is failing. When I run linters locally, it says that there are no outstanding changes:

```
➜  aqueduct git:(eng-2620-dag-line-crossing-fix) python3 scripts/run_linters.py --golang
Linting Go code...
WARN [linters context] rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] sqlclosecheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] wastedassign is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
➜  aqueduct git:(eng-2620-dag-line-crossing-fix) git status
On branch eng-2620-dag-line-crossing-fix
Your branch is up to date with 'origin/eng-2620-dag-line-crossing-fix'.

nothing to commit, working tree clean
```

**Comment by vsreekanti** [comment]
Makes sense! I'll let @likawind comment on the implementation because he knows the code better than I do. 

re: linting, is your linter up to date with what's on GitHub Actions?

**Review by likawind — APPROVED** [review]
The code LGTM! @vsreekanti the positioning algorithm takes operator / artifacts as input, so it will be some work to take account collapsed metrics / checks. Given the current solution solves most problem, I think we can do the collapsing in a separatly?

**Comment by likawind** [comment]
@agiron123 could you help removing the `var err error = nil` line and see if the GH action still complaints? I don't know why this is not caught before but the warning seems to be reasonable

**Comment by agiron123** [comment]
> @agiron123 could you help removing the `var err error = nil` line and see if the GH action still complaints? I don't know why this is not caught before but the warning seems to be reasonable

@vsreekanti Looks like I was on version 1.48, when the latest version for golangci-lint is 1.52.

**Comment by vsreekanti** [comment]
Yeah, that's a bit of a pain sometimes — that's usually the first thing to check if linters don't work.

**Comment by vsreekanti** [comment]
Hm, it looks like we merged this without addressing the challenges around node spacing. Have we created a separate task for tracking that?

**Comment by agiron123** [comment]
> Hm, it looks like we merged this without addressing the challenges around node spacing. Have we created a separate task for tracking that?

Sorry, I thought @likawind 's comment below addressed that we should be doing this in another PR:
The code LGTM! @vsreekanti the positioning algorithm takes operator / artifacts as input, so it will be some work to take account collapsed metrics / checks. Given the current solution solves most problem, I think we can do the collapsing in a separatly?

I've created a Linear task here to track:
https://linear.app/aqueducthq/issue/ENG-2692/dag-node-spacing

**Comment by vsreekanti** [comment]
Awesome, thanks for tracking it!

**Pull request merged**